crates/test_parallel_contract/src/lib.rs

```diff
@@ -112,7 +113,7 @@ impl TestContract {
                         let args = CKDArgs {
                             request: CKDRequestArgs {
                                 domain_id: *domain_id,
-                                app_public_key: generate_app_public_key(i as u128 + 2),
+                                app_public_key: generate_app_public_key((seed + i + 2) as u128),
```

Comment:
While we don't have the same requirements for the devnet code, just as a good habit I'd suggest:

```suggestion
                                app_public_key: generate_app_public_key(u128::from(seed + i + 2)),
```